src/Native/Runtime/unix/unixasmmacrosamd64.inc

```diff
@@ -238,7 +238,17 @@ C_FUNC(\Name):
 #define THREAD__VALIDATEEXINFOPOP C_FUNC(_ZN6Thread17ValidateExInfoPopEP6ExInfoPv)
 
 .macro INLINE_GETTHREAD
-        call    C_FUNC(RhpGetThread)
+        // Inlined version of call C_FUNC(RhpGetThread)
+       .att_syntax
+#if defined(__APPLE__)
+        movq    _tls_CurrentThread@TLVP(%rip), %rdi
+        callq   *(%rdi)
+#else
+        leaq    tls_CurrentThread@TLSLD(%rip), %rdi
```

Comment:
The ELF loader will patch this with mov from fs and no-ops. There is no actual call here if you step through this code when it is statically linked into main executable. We could have a separate build flavors, but I doubt that it is worth it to save a few no-ops.